webboot/webboot.go

```diff
@@ -12,20 +12,46 @@ import (
 	"log"
 
 	"github.com/u-root/webboot/pkg/dhclient"
+	"github.com/u-root/webboot/pkg/mountkexec"
+	"github.com/u-root/webboot/pkg/webboot"
 )
 
 var (
-	ifName  = flag.String("interface", "^e.*", "Name of the interface")
-	timeout = flag.Int("timeout", 15, "Lease timeout in seconds")
-	retry   = flag.Int("retry", 5, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
-	verbose = flag.Bool("verbose", false, "Verbose output")
-	ipv4    = flag.Bool("ipv4", true, "use IPV4")
-	ipv6    = flag.Bool("ipv6", true, "use IPV6")
+	cmd       = flag.String("cmd", "", "Command Line")
+	mountDir  = flag.String("dir", "/tmp/mountDir", "The mount point of the ISO")
+	ifName    = flag.String("interface", "^e.*", "Name of the interface")
+	timeout   = flag.Int("timeout", 15, "Lease timeout in seconds")
+	retry     = flag.Int("retry", 5, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
+	verbose   = flag.Bool("verbose", false, "Verbose output")
+	ipv4      = flag.Bool("ipv4", true, "use IPV4")
+	ipv6      = flag.Bool("ipv6", true, "use IPV6")
+	osSystems = map[string]webboot.Distro{
+		"TinyCore": webboot.Distro{"/boot/vmlinuz64 ", "/boot/initrd", "--reuse-cmdline", "https:louis.com"},
+		"Ubuntu":   webboot.Distro{"/boot/vmlinuz ", "/boot/init", "--reuse-cmdline", "https:louis.com"},
+	}
 )
 
 func main() {
+
 	flag.Parse()
 
 	dhclient.Request(*ifName, *timeout, *retry, *verbose, *ipv4, *ipv6)
-	log.Println("weboot up")
+
+	if len(flag.Args()) < 1 {
+		log.Fatal("error:pass in an operating system")
+	}
+
+	osname := flag.Args()[0]
+
+	if _, ok := osSystems[osname]; !ok {
```

Comment:
you'll need to talk to urvisha about this, how do we work when people give you a valid URL.

Reply:
Urvisha got it working on her branch and we should be able to get it in on the next PR.